webapp/src/filework/mod.rs

```diff
@@ -1,35 +1,70 @@
-#![allow(unused)]
-use std::fs::{File, remove_file};
-use std::io::{Write, Seek, SeekFrom};
-use std::env;
-use std::path::{PathBuf, Path};
-
-pub fn write_source_to_file(source_code: &str, lang: &str) -> PathBuf
-{
-    // If error ever happens on work with temp files it's not on the user, 
-    // so he should get "internal server error" here
-    let mut input_file_name = env::var("COMPILATION_TEMP_DIR")
-        .expect("COMPILATION_TEMP_DIR doesn't exist");
-    input_file_name.push_str("/");
-    input_file_name.push_str(
-        &chrono::Utc::now()
-        .format("compilation_input-%Y-%m-%d-%H-%M-%S-%f").to_string()
-    );
-    input_file_name.push_str(".cpp");     // TODO: add lang specific extensions
-
-    let mut code_file = File::create(&input_file_name)
-        .expect("Panic on temp file creation");  
-    code_file.write_all(source_code.as_bytes())
-        .expect("Panic on copying of source code into the temp file");
-    code_file.seek(SeekFrom::Start(0)).expect("Panic on temp file seek");
-    
-    drop(code_file);
-
-    PathBuf::from(input_file_name)
-}
-
-pub fn delete_file(filename: &Path)
-{
-    remove_file(filename)
-        .expect("Panic on file deletion");
-}
+#![allow(unused)]
+use std::fs::{File, remove_file};
+use std::io::{Write, Seek, SeekFrom};
+use std::env;
+use std::path::{PathBuf, Path};
+use std::error::Error;
+
+pub fn write_source_to_file(source_code: &str, lang: &str) -> Option<PathBuf>
+{
+    // If error ever happens on work with temp files it's not on the user, 
+    // so he should get "internal server error" here
+    let mut input_file_name = env::var("COMPILATION_TEMP_DIR")
+        .expect("COMPILATION_TEMP_DIR doesn't exist");
+    input_file_name.push_str("/");
+    input_file_name.push_str(
+        &("compilation_input-".to_owned() + &generate_file_signature())
+    );
+    input_file_name.push_str(".cpp");     // TODO: add lang specific extensions
+
+    let mut code_file: File;
+    match File::create(&input_file_name)
+    {
+        Ok(file) => code_file = file,
+        Err(_) =>
+        {
+            println!("[ERROR]: Couldn't create a file at \"{}\"", input_file_name);
+            return None;
+        }
+    }
+        
+    match code_file.write_all(source_code.as_bytes())
+        .and_then(|()| code_file.seek(SeekFrom::Start(0)))
+    {
+        Ok(_) => {},
+        Err(_) =>
+        {
+            println!("[ERROR]: Error while working with \"{}\"", input_file_name);
+            return None;
+        }
+    }
+
+    println!("[INFO]: Created \"{}\"", input_file_name);
+    Some(PathBuf::from(input_file_name))
+}
+
+pub fn delete_file(filename: &Path)
+{
+    match remove_file(filename)
+    {
+        Ok(_) => println!("[INFO]: Removed {:?}.", filename),

```

Comment:
Can you create a macro `info!("removed {}", filename)`?
It can unwrap to something like `println!("[info] removed {} (filework/mod.rs:50)`

Reply:
Sure. Looks like it's a #9 issue, is it okay if it'll be worked on as part of #9? And left as is for this pull request.